mod.d/enadiag.yaml

```diff
@@ -0,0 +1,270 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+--- !ec2rlcore.module.Module
+# Module document. Translates directly into an almost-complete Module object
+name: !!str enadiag
+path: !!str
+version: !!str 1.0
+title: !!str Checks the ethool -S output for ENA specific statistics to diagnose issues
+helptext: !!str |
+  Checks for ENA specific statistics on link flaps, throughput/packets per second overages, etc.
+placement: !!str run
+package:
+  - !!str
+language: !!str python
+remediation: !!str False
+content: !!str |
+  from __future__ import print_function
+  from distutils.version import LooseVersion
+
+  import os
+  import re
+  import subprocess
+  import sys
+
+  class Interface:
+      def __init__(self,
+                   name="",
+                   tx_timeout=0,
+                   suspend=0,
+                   wd_expired=0,
+                   interface_down=0,
+                   admin_q_pause=0,
+                   bw_in_allowance_exceeded=0,
+                   bw_out_allowance_exceeded=0,
+                   pps_allowance_exceeded=0,
+                   conntrack_allowance_exceeded=0,
+                   linklocal_allowance_exceeded=0):
+
+          self.name = name
+          self.tx_timeout = tx_timeout
+          self.suspend = suspend
+          self.wd_expired = wd_expired
+          self.interface_down = interface_down
+          self.admin_q_pause = admin_q_pause
+          self.bw_in_allowance_exceeded = bw_in_allowance_exceeded
+          self.bw_out_allowance_exceeded = bw_out_allowance_exceeded
+          self.pps_allowance_exceeded = pps_allowance_exceeded
+          self.conntrack_allowance_exceed = conntrack_allowance_exceeded
+          self.linklocal_allowance_exceeded = linklocal_allowance_exceeded
+
+  def get_interfaces():
+      """
+      Return a list of non-loopback interfaces on the system.
+
+      Returns:
+          interface_list (list): List of interfaces (e.g. ["eth0", "eth1"]
+      """
+      try:
+          interface_list = []
+          # Get all net interfaces except loopback, append to list, sort it.
+          for interface in os.listdir("/sys/class/net"):
+              if interface != "lo":
+                  interface_list.append(interface)
+          if len(interface_list) > 0:
+              interface_list = sorted(interface_list)
+          # Return an error if the list is somehow empty (didn't find any network interfaces)
+          else:
+              raise ValueError("No interfaces found.")
+      # Catch OSError in Python2 and FileNotFoundError in Python3 (inherits from OSError)
+      except OSError:
+          print("Unable to build interface list.")
+      return interface_list
+
+  # Create list with get_interfaces and create Interface class objects with them.
+  interfaces = get_interfaces()
```

Comment:
These seem out of place. Could they be moved into run()?

Reply:
Yes - I actually thought I already had when moving from my initial PoC, but I guess not!